model/src/main/scala/org/broadinstitute/dsde/rawls/model/WorkspaceModel.scala

```diff
@@ -187,11 +190,46 @@ case class EntityCopyDefinition(
                    entityNames: Seq[String]
                    )
 
+// In the next phase (GAWB-3100), this class's members will be swapped out for a singular methodUri
 case class MethodRepoMethod(
                    methodNamespace: String,
                    methodName: String,
                    methodVersion: Int
-                   )
+                   ) {
+  def asAgoraMethodUrl: String = asMethodUrlForRepo("agora")
+
+  // Next phase: this method goes away
+  def asMethodUrlForRepo(repository: String): String = {
+
+    def existsAndNonEmpty(subject: String) = subject != null && subject.nonEmpty
+
+    (existsAndNonEmpty(repository),
+     existsAndNonEmpty(methodNamespace),
+     existsAndNonEmpty(methodName),
+     methodVersion > 0) match {
+      case (true, true, true, true) => s"$repository://$methodNamespace/$methodName/$methodVersion"
+      case _ => throw new RawlsException(
+        s"Could not generate a method URI from MethodRepoMethod with repo \'$repository\', namespace \'$methodNamespace\', name \'$methodName\', version \'$methodVersion\'"
+      )
+    }
+  }
+}
+
+object MethodRepoMethod {
+  def apply(uri: String): MethodRepoMethod = {
+
+    (for {
+      parsedUri <- Try(parse(uri)).toOption
+      namespace <- parsedUri.host
+      parts     <- Option(parsedUri.pathParts)
+      name      <- Option(parts.head.part)
+      version   <- Try(parts(1).part.toInt).toOption
+      result    <- if (parts.size == 2) Some(MethodRepoMethod(namespace, name, version)) else None
+    } yield {
+      result
+    }).getOrElse(throw new RawlsException(s"Could not create a MethodRepoMethod from URI \'$uri\'"))
+  }
```

Comment:
I think it would be better to throw an exception here in the `apply` instead of possibly creating a `MethodRepoMethod` with None/nulls in it - then, you can avoid the null check above.

Reply:
Good call, made this tweak.